lib/Params.ml

```diff
@@ -328,6 +328,14 @@ module Exp = struct
       | Str _ | Lb _ | Exp {pexp_desc= Pexp_ifthenelse _ | Pexp_let _; _} ->
           false
       | _ -> true
+
+  let ifthenelse_inner_pro (c : Conf.t) ~parens ~ctx0 =
+    if parens then false
+    else if Stdlib.(c.fmt_opts.if_then_else.v = `Fit_or_vertical) then
+      match ctx0 with
+      | Exp {pexp_desc= Pexp_beginend _; _} -> true
```

Comment:
Did you add this special case because you use the `if_then_else` option ? I'd really like to remove any special case here. I think it's better to change how `if_then_else` is implemented and to put the pro at the inner position without special cases.

Reply:
No, it was just to fix the regression that only showed up with fit-or-vertical. I noticed this regression was because when you had an `if` inside another, `pro` would be used.

Reply:
I much prefer the new version! It makes an existing special case explicit instead of adding an other one on top. This will help whoever wants to refactor how ifthenelse are formatted in the future.